### PR TITLE
[model] feat: support npu qwen3p5 and qwen3p5-moe

### PR DESCRIPTION
**Pull request by yicheng-gong**
### What does this PR do?

> Concise overview of the change. Reference related issues/PRs.

### Checklist Before Starting

- Search for relative PRs/issues and link here: ...
- PR title follows `[{modules}] {type}: {description}` format (see [check_pr_title.yml](.github/workflows/check_pr_title.yml) for the full list of allowed modules and types)
  - Breaking changes: prepend `[BREAKING]` — e.g. `[BREAKING][parallel, model] feat: dynamic batching`

### Test

> Validation results (training curves, eval metrics) for changes not covered by CI.

### API and Usage Example

> Show API changes and usage examples if applicable.

### Design & Code Changes

> High-level design description and specific change list.

### Checklist Before Submitting

- Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md)
- Applied pre-commit checks
- Added/updated documentation
- If `tasks/` training scripts were moved or renamed: updated `docs/` examples and verified `python3 scripts/ci/check_doc_task_paths.py` passes (also enforced by the **Check doc task paths** CI workflow)
- Added tests to CI workflow (or explained why not feasible)

### CODE REVIEW

File: veomni/models/transformers/qwen3_5/generated/patched_modeling_qwen3_5_npu.diff [hunk 1]
```diff
@@ -0,0 +1,1319 @@
...
++        # via all_gather to compute them locally.
++        image_mask = kwargs.get("image_mask", None)
++        video_mask = kwargs.get("video_mask", None)
++
++        # if None, calculate mask
++        if video_mask is None and image_mask is None:
++            if get_parallel_state().sp_enabled:
++                input_ids_list = [torch.zeros_like(input_ids) for i in range(get_parallel_state().sp_size)]
++                dist.all_gather(input_ids_list, input_ids, group=get_parallel_state().sp_group)
++                input_ids = torch.cat(input_ids_list, dim=0)
```
[the remark this answers is not in the document]
Comment (yicheng-gong) [reply veomni/models/transformers/qwen3_5/generated/patched_modeling_qwen3_5_npu.diff:976]: same as gpu, with no other changes

File: veomni/models/transformers/qwen3_5_moe/generated/patched_modeling_qwen3_5_moe_npu.diff [hunk 2]
```diff
@@ -0,0 +1,1499 @@
...
++        image_mask = kwargs.get("image_mask", None)
++        video_mask = kwargs.get("video_mask", None)
++
++        # if None, calculate mask
++        if video_mask is None and image_mask is None:
++            if get_parallel_state().sp_enabled:
++                input_ids_list = [torch.zeros_like(input_ids) for i in range(get_parallel_state().sp_size)]
++                dist.all_gather(input_ids_list, input_ids, group=get_parallel_state().sp_group)
++                input_ids = torch.cat(input_ids_list, dim=0)
++            image_mask, video_mask = self.get_placeholder_mask(input_ids)
```
[the remark this answers is not in the document]
Comment (yicheng-gong) [reply veomni/models/transformers/qwen3_5_moe/generated/patched_modeling_qwen3_5_moe_npu.diff:1138]: same as gpu, with no other changes

File: veomni/models/transformers/qwen3_5_moe/generated/patched_modeling_qwen3_5_moe_npu.diff [hunk 3]
```diff
@@ -0,0 +1,1499 @@
...
+@@ -1814,7 +2215,19 @@
+                 inputs_embeds=inputs_embeds,
+                 attention_mask=attention_mask,
+                 past_key_values=past_key_values,
++                mm_token_type_ids=mm_token_type_ids,
+             )
++        else:
++            # --- Patch.3: Transpose pre-computed position_ids if they follow VeOmni collation format ---
++            # When position_ids are pre-computed during data preprocessing (for varlen/packed data),
++            # they are typically collated into (batch_size, 3, seq_len) shape. We transpose them
```
[the remark this answers is not in the document]
Comment (yicheng-gong) [reply veomni/models/transformers/qwen3_5_moe/generated/patched_modeling_qwen3_5_moe_npu.diff:1275]: same as gpu, with no other changes

**Review by FoolPlayer — APPROVED** [review]
